Fix: hystartup affects tests

allison-casey opened the pull request:
closes #2029 also fixes macros not being able to find the hystartup module causing it to error out.

Kodiologist commented:
The changes sound good, but the commit message doesn't seem to match the changes. I think that `os.environ.pop("HYSTARTUP", None)` is the only line related to the commit message. Also, you can write that as `del os.environ['HYSTARTUP']`, right?

allison-casey commented:
`del os.environ ['HYSTARTUP']` will `KeyError` if the key doesn't exist so the `pop` is easier than an if guard. 

Changed the message to be more generally about adding `HYSTARTUP` to tests and moved the fix for macros failing to expand in the `HYSTARTUP` module. which i forgot to mention i found while adding unit tests.

Kodiologist commented:
> `del os.environ ['HYSTARTUP']` will `KeyError` if the key doesn't exist

Huh, I learn a lot of things that are dumb about Python's semantics by working on Hy, but that's a new one.
